i18n/src/parser.js

```diff
@@ -33,7 +46,12 @@ function parsePattern(pattern) {
       positive = patternParts[0],
       negative = patternParts[1];
 
-  var positiveParts = positive.split(DECIMAL_SEP),
+  // The parsing logic below assumes that there will always be a DECIMAL_SEP in the pattern.
```

Comment:
Pretty much. We add it for the sake of parsing, because what comes after the DECIMAL_SEP determines the min/max fraction size (depending on the number of ZERO and DIGIT occurrences) and also the `posSuf`. For example, if we have `#,##0$`, we convert it to `#,##0.$`, which (when parsed) results in `minFrac: 0`, `maxFrac: 0`, `posSuf: $` - i.e. have 0 fraction digits and tput `$` after the number (`posSuf`).